src/hooks/shards/useShardsList.ts

```diff
@@ -34,10 +52,10 @@ const useShardsList = (catalogNames: string[]) => {
 
     const swrKey = useMemo(
         () =>
-            session && catalogNames.length > 0
-                ? `shards-${catalogNames.join('-')}`
+            session && taskAuthorizationData?.length && catalogNames.length > 0
```

Comment:
Asked a question in convo so marking answer here.

SWR will serialize the array passed in when generating the cache key on their end.